File: app/src/main/java/com/quran/labs/androidquran/widgets/QuranViewPager.java
package com.quran.labs.androidquran.widgets;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class QuranViewPager extends ViewPager {

  public QuranViewPager(Context context) {
    super(context);
  }

  public QuranViewPager(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  @Override
  public boolean onTouchEvent(MotionEvent ev) {
    try {
      return super.onTouchEvent(ev);
    } catch (IllegalArgumentException e) {
      return false;
    }
  }
}

File: app/src/main/java/com/quran/labs/androidquran/view/QuranViewPager.kt
package com.quran.labs.androidquran.view

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class QuranViewPager : ViewPager {

  constructor(context: Context) : super(context)

  constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

  override fun onTouchEvent(ev: MotionEvent): Boolean {
    return try {
      super.onTouchEvent(ev)
    } catch (e: IllegalArgumentException) {
      false
    }
  }
}
